server/models/Article.js

```diff
@@ -1,9 +1,17 @@
 const mongoose = require('mongoose');
+const CategoryModel = require('../models/Category');
 const { Schema } = mongoose;
 
 const schema = new Schema({
     title: String,
-    fullContent: String
+    fullContent: String,
+    categoryId: [{
+        text: String,
+        postedBy: {
```

Comment:
why do you need postedBy?

Reply:
and it's not an array either - the article belongs to one category so something like 
```js
categoryId: {
   type: ...
   ref: ....
}
```
should be enough